VirtoCommerce.Storefront/Domain/ContentBlobProviders/JunctionPoint.cs

```diff
@@ -0,0 +1,420 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+namespace VirtoCommerce.Storefront.Domain.ContentBlobProviders
+{
+    /// <summary>
+    /// Provides access to NTFS junction points in .Net.
+    /// </summary>
+    public static class JunctionPoint
```

Comment:
Move all P/Invoke code to separate class and mark it with [SecurityCritical]. Mark this class as SecuritySafeCritical